model/Aktivita/Aktivita.php

```diff
@@ -1013,12 +1018,15 @@ public static function uloz(
         string  $obrazekSoubor = null,
         string  $obrazekUrl = null,
         int     $odmenaZaHodinu = null,
+        ?bool    $maPravoNaProvadeniKorekci = null,
     ): Aktivita {
         $data[Sql::BEZ_SLEVY]    = (int)!empty($data[Sql::BEZ_SLEVY]);    // checkbox pro "bez_slevy"
         $data[Sql::NEDAVA_BONUS] = (int)!empty($data[Sql::NEDAVA_BONUS]); // checkbox pro "nedava_bonus"
-        $nastavujeKorekci = empty($data[Sql::ID_AKCE]) /* nová aktivita */
-            || array_key_exists(Sql::PROBEHLA_KOREKCE, $data);
-        if (array_key_exists(Sql::PROBEHLA_KOREKCE, $data)  /** editace korekce; reakce na změnu textu viz @see popis */
+        $nastavujeKorekci        = empty($data[Sql::ID_AKCE]) /* nová aktivita */
+                                   || $maPravoNaProvadeniKorekci
+                                   || ($maPravoNaProvadeniKorekci === null && array_key_exists(Sql::PROBEHLA_KOREKCE, $data));
```

Comment:
`array_key_exists(Sql::PROBEHLA_KOREKCE, $data)` je chyták, minimálně pro Firefox - když je checkbox odškrtnutý, tady ve významu "ruším potvrzení předchozí korektury", tak se na BE neodešle vůbec .

Takže odškrtnutý checkbox nikdy v `$data` není a my z dat nepoznáme, jestli se checkbox vůbec nezobrazil, nebo ho uživatel odškrtnul.

Navrhuju to vyřešit přes práva - když uživatel má právo na korekturu, tak se mu určitě checbox zobrazil a já můžu brát `empty($data[Sql::PROBEHLA_KOREKCE])` jako odškrtnutí.